microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/utils/AnnotationUtils.java

```diff
@@ -281,7 +283,68 @@ public static boolean isMatchAnnotation(Annotation annotation, String annotation
 	public static String getAnnotationMemberValue(IAnnotation annotation, String memberName) throws JavaModelException {
 		for (IMemberValuePair pair : annotation.getMemberValuePairs()) {
 			if (memberName.equals(pair.getMemberName())) {
-				return pair.getValue() != null ? pair.getValue().toString() : null;
+				String memberValue = pair.getValue() != null ? pair.getValue().toString() : null;
+				if (memberValue != null) {
+					switch (pair.getValueKind()) {
+					case IMemberValuePair.K_STRING:
+						// ex : @Path("foo") --> returns foo
+						return memberValue;
+					case IMemberValuePair.K_SIMPLE_NAME: {
+						// ex : private static final String PATH = "foo";
+						// @Path(PATH) --> returns foo
+						IType type = ((IType) annotation.getAncestor(IJavaElement.TYPE));
+						if (type != null) {
+							String fieldValue = getFieldValue(type, memberValue);
+							if (fieldValue != null) {
+								return fieldValue;
+							}
+						}
+						return memberValue;
+					}
+					case IMemberValuePair.K_QUALIFIED_NAME: {
+						// ex : class SomePage {
+						// private static final String PATH = "foo";
+						// @Path(SomePage.PATH) --> returns foo
+						IType type = ((IType) annotation.getAncestor(IJavaElement.TYPE));
+						if (type != null) {							
+							int index = memberValue.lastIndexOf(".");
+							String fieldName = memberValue.substring(index + 1);
+							String valueClass = memberValue.substring(0, index);
+							String[][] resolvedTypes = type.resolveType(valueClass);

```

Comment:
I guess the only other question I would have had is whether it's necessary to call `resolveType` here. Can't you pass `valueClass` directly to `findType(..)` from below ? It should be a fully qualified type after all.